Fix detection whether filename needs to be truncated (tilde in the middle)
in dialog boxes, such as copy file.

diff -Naurdp mc-4.6.1.orig/src/file.c mc-4.6.1/src/file.c
--- mc-4.6.1.orig/src/file.c	2005-09-05 17:08:08.000000000 +0200
+++ mc-4.6.1/src/file.c	2005-09-05 17:18:22.000000000 +0200
@@ -1704,13 +1704,13 @@ panel_operate_generate_prompt (const WPa
     *dp = '\0';
 
     if (single_source) {
-	i = fmd_xlen - strlen (format_string) - 4;
+	i = fmd_xlen - mbstrlen (format_string) - 4;
 	g_snprintf (cmd_buf, sizeof (cmd_buf), format_string,
 		    name_trunc (single_source, i));
     } else {
 	g_snprintf (cmd_buf, sizeof (cmd_buf), format_string,
 		    panel->marked);
-	i = strlen (cmd_buf) + 6 - fmd_xlen;
+	i = mbstrlen (cmd_buf) + 6 - fmd_xlen;
 	if (i > 0) {
 	    fmd_xlen += i;
 	    fmd_init_i18n (TRUE);	/* to recalculate positions of child widgets */
